test/benchmarks/Benchmarks.Trace/SerilogBenchmark.cs

```diff
@@ -0,0 +1,79 @@
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.Configuration;
+using Serilog;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class SerilogBenchmark
+    {
+        private static readonly Logger EnrichedLogger;
+        private static readonly Logger Logger;
+        private static readonly Tracer LogInjectionTracer;
+        private static readonly Tracer BaselineTracer;
+
+        static SerilogBenchmark()
+        {
+            var logInjectionSettings = new TracerSettings
+            {
+                StartupDiagnosticLogEnabled = false,
+                LogsInjectionEnabled = true,
+                Environment = "env",
+                ServiceVersion = "version"
+            };
+
+            LogInjectionTracer = new Tracer(logInjectionSettings, new DummyAgentWriter(), null, null, null);
+            Tracer.Instance = LogInjectionTracer;
+
+            var baselineSettings = new TracerSettings
+            {
+                StartupDiagnosticLogEnabled = false,
+                LogsInjectionEnabled = false,
+                Environment = "env",
+                ServiceVersion = "version"
+            };
+
+            BaselineTracer = new Tracer(baselineSettings, new DummyAgentWriter(), null, null, null);
+
+            EnrichedLogger = new LoggerConfiguration()
```

Comment:
So `.Enrich.FromLogContext()` is currently a requirement for logs injection to work, but should we also measure a third case where `.Enrich.FromLogContext()` is enabled but our logs injection isn't added? That would help isolate the impact of logs injection

Reply:
That's fair, added in https://github.com/DataDog/dd-trace-dotnet/pull/1351/commits/7e040c75fe1694851165147d0a60030c68214b89

Interestingly it doesn't seem to make a difference in the results. Looks like most of the stuff is lazy, so the first property added to the context makes the biggest impact.